### PR TITLE
Add 'current' url version alias

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #384

### CODE REVIEW

**Comment by diekhans** [comment]
+1

**Comment by saupchurch** [comment]
Do we expect the reference server handle multiple simultaneous versions of the schema?  I'm +1 for implementing 'current' URL for now to make development and demo work easier (although the implementation is a bit more 'ignore version' than 'use the current version'), but perhaps at some point we should consider moving the version number from the URL to the result JSON and stabilize the URL.

**Comment by dcolligan** [comment]
@saupchurch we only handle a single version right now.  Handling multiple versions would be a lot of work (but may be something that happens in the distant future).

**Comment by diekhans** [comment]
Give how hard it is to get the resource to do one version, I
would say that multiple version should not be an immediate goal.

**Comment by saupchurch** [comment]
If there is only a single supported version then it seems unnecessary to have it in the URL at all as it can either be gleaned from the code or reported statically on the server home page.  Making the URL dependent on version just breaks things when it gets changed which is what drives us to replace it with 'current' in the first place.

**Comment by diekhans** [comment]
The version does change, so it's an indication that it has.

Not all server instances running will be using the same version.

It's good planning for the future.

It is also an example of how version support can be done
for other servers.

**Comment by saupchurch** [comment]
OK +1

**Comment by dcolligan** [comment]
We want an easy, programatic way to indicate to the client that they're using the wrong version before they get some low-level error that is unintelligible to them.  Using `current` is easier to use, but also foregoes this check, so really should only be used be people that know what they're doing (i.e. what version they are on).

File: ga4gh/frontend.py [hunk 1]
```diff
@@ -228,7 +230,9 @@ def handleFlaskPostRequest(version, flaskRequest, endpoint):
     at at the specified version. Invokes the specified endpoint to
     generate a response.
     """
-    if Version.parseString(version) != Version.parseString(protocol.version):
+    if version == Version.currentString:
```
Comment (jeromekelleher) [inline ga4gh/frontend.py]: I think this logic should be internalised into the Version class. What if we had a method on version called `isCurrentVersion` which takes a string, which can be either 'current' or a version string and returns a Boolean?

**Comment by jeromekelleher** [comment]
I'm +1 for this change (modulo the nitpicks above). 

However, I think we should be using 'Content-Type' as the definitive API versioning mechanism, as the Matchmaker team have done. They use a content-type which is something like 'json+mme-v1.0' (sorry, I'm doing this from memory, and can't find the slides!), which is far better than a URL prefix I think. Among other advantages, this allows the client and server negotiate a protocol version that they both understand and seems a lot more flexible than URLs.

Anyway, this would need to be specified on the schemas repo before it should be implemented here, so it's not directly relevent to this PR.

**Comment by diekhans** [comment]
Suggest investigating if this Content-Type approach is used
elsewhere.  It's appealing, although I am a bit worried about
unexpected downsides.  Generally content type is fairly generic
format.

**Comment by diekhans** [comment]
whoever designed &format=json doesn't understand HTTP

**Comment by pgrosu** [comment]
You prefer to define a new protocol - I believe `JSTP` and `JTP` are taken :)  I understand but it's pretty much the standard these days with many APIs over the web.

**Comment by jmarshall** [comment]
> I think we should be using 'Content-Type' as the definitive API versioning mechanism

:+1:

The first page of Google search results for `content-type versioning` make for good reading.

**Comment by dcolligan** [comment]
If anyone wants to file an issue on schemas for content-type versioning (although I find the idea a little strange myself... seems like a higher bar for custom clients to implement), be my guest.  This change is about an incremental improvement in the URL versioning scheme.

**Comment by jeromekelleher** [comment]
Agreed @dcolligan --- sorry for veering off on a tangent there.

**Comment by dcolligan** [comment]
@jeromekelleher fixed nit

**Comment by diekhans** [comment]
ticket for collection HTTP/ReST specification issue is here:

       https://github.com/ga4gh/schemas/issues/300

**Comment by jeromekelleher** [comment]
Thanks @dcolligan, merging.

**Pull request merged**